### DISCUSSION TITLE
v0.1.0-beta.3

### DISCUSSION THREAD

**Discussion by ChuxiJ** [ace-step/ACE-Step-1.5#544] [Announcements]
## What's Changed
* docs: add AGENTS.md guardrails for AI-assisted contributions by @1larity in https://github.com/ace-step/ACE-Step-1.5/pull/475
* Fix: {TRACK_NAME} not replaced by actual track name in API. by @goedzo in https://github.com/ace-step/ACE-Step-1.5/pull/474
* fix: seeds param by @DumoeDss in https://github.com/ace-step/ACE-Step-1.5/pull/496
* refactor(handler part 4): extract diffusion mixin and add validation tests by @1larity in https://github.com/ace-step/ACE-Step-1.5/pull/490
* Fix repaint mode not populating lyrics from generated audio metadata by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/489
* [Fix] Eliminate redundant LLM load/offload cycles in PT batch mode by @agorevski in https://github.com/ace-step/ACE-Step-1.5/pull/492
* Fix LoRA memory bloat: replace deepcopy with CPU state_dict backup by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/499
* fix: phase-aware max_new_tokens to fix misleading progress bar stopping early by @agorevski in https://github.com/ace-step/ACE-Step-1.5/pull/493
* Add CodeQL analysis workflow configuration by @schneidergithub in https://github.com/ace-step/ACE-Step-1.5/pull/502
* Use CPU for LM by @Jay4242 in https://github.com/ace-step/ACE-Step-1.5/pull/497
* Add minimal Copilot instructions for repository by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/506
* Feat update skills simplemv by @DumoeDss in https://github.com/ace-step/ACE-Step-1.5/pull/500
* Fix YAML indentation in CodeQL workflow configuration by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/503
* refactor(handler part  5+6 ): Extend decomposition and harden source-audio analyze validation by @1larity in https://github.com/ace-step/ACE-Step-1.5/pull/508
* fix(handler): address post-merge review feedback in io/prompt/task/padding mixins by @1larity in https://github.com/ace-step/ACE-Step-1.5/pull/510
* Potential fix for code scanning alert no. 35: Uncontrolled data used in path expression by @schneidergithub in https://github.com/ace-step/ACE-Step-1.5/pull/507
* fix: improve error handling in training and generation handlers by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/513
* Doc only: Give coderabbit explicit guidance on module loc limits. by @1larity in https://github.com/ace-step/ACE-Step-1.5/pull/512
* Demote vllm→PyTorch backend auto-selection from warning to info by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/511
* feat: add dgx spark support by @NicasioSirvent in https://github.com/ace-step/ACE-Step-1.5/pull/485
* support cover (exp) by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/479
* feat: add Side-Step training v2 (corrected LoRA fine-tuning) by @koda-dernet in https://github.com/ace-step/ACE-Step-1.5/pull/478
* Revert "feat: add dgx spark support" by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/514
* Skip torch.distributed initialization in single-GPU mode by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/477
* Fix CodeQL alert #59: Sanitize checkpoint paths to prevent deserialization attacks by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/525
* Streamline path traversal security fix: remove verbose logging and defensive checks by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/526
* fix for intel gpu support by @xushengyuan in https://github.com/ace-step/ACE-Step-1.5/pull/515
* Refactor(handler part 7): continue FD extraction for batch conditioning and embedding pipeline by @1larity in https://github.com/ace-step/ACE-Step-1.5/pull/516
* fix(training): guard pin_memory_device when None to prevent DataLoader crash by @jayvenn21 in https://github.com/ace-step/ACE-Step-1.5/pull/517
* (Feat)(mlx): Add dit progress bar and Fix the final-step `break  by @tonyjohnvan in https://github.com/ace-step/ACE-Step-1.5/pull/519
* Potential fix for code scanning alert no. 59: Deserialization of user-controlled data by @schneidergithub in https://github.com/ace-step/ACE-Step-1.5/pull/524
* Support local LoKr safetensors in adapter loader by @riversedge in https://github.com/ace-step/ACE-Step-1.5/pull/530
* (feat)(api)Enhance audio result data structure to include per-audio metadata by @tonyjohnvan in https://github.com/ace-step/ACE-Step-1.5/pull/531
* Address review feedback: remove redundant path validations and tighten exception handling by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/532
* Potential fix for code scanning alert no. 59: Deserialization of user-controlled data by @schneidergithub in https://github.com/ace-step/ACE-Step-1.5/pull/529
* Polishing tooltips in Gradio to de-clutter the GUI by @pedro16797 in https://github.com/ace-step/ACE-Step-1.5/pull/527
* Refactor tooltip implementation in Gradio to improve GUI clarity and … by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/534
* fix: vocal language ui by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/535
* (feat)(mlx): Supports MLX Pre-compile model for DiT  by @tonyjohnvan in https://github.com/ace-step/ACE-Step-1.5/pull/522
* feat: enhances the Extract mode  by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/536
* fix: only hide optional outputs in Extract mode, not all modes by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/537
* fix: Lego mode UI redesign and Extract/Lego mode switch cleanup by @DumoeDss in https://github.com/ace-step/ACE-Step-1.5/pull/538
* feat: add inline help buttons with modal documentation to Gradio UI by @ChuxiJ in https://github.com/ace-step/ACE-Step-1.5/pull/542
* Fix nanovllm CUBLAS error on Turing GPUs by detecting bfloat16 support by @Copilot in https://github.com/ace-step/ACE-Step-1.5/pull/541

## New Contributors
* @schneidergithub made their first contribution in https://github.com/ace-step/ACE-Step-1.5/pull/502
* @Jay4242 made their first contribution in https://github.com/ace-step/ACE-Step-1.5/pull/497
* @NicasioSirvent made their first contribution in https://github.com/ace-step/ACE-Step-1.5/pull/485
* @koda-dernet made their first contribution in https://github.com/ace-step/ACE-Step-1.5/pull/478
* @pedro16797 made their first contribution in https://github.com/ace-step/ACE-Step-1.5/pull/527

**Full Changelog**: https://github.com/ace-step/ACE-Step-1.5/compare/v0.1.0-beta.2...v0.1.0-beta.3

<hr /><em>This discussion was created from the release <a href='https://github.com/ace-step/ACE-Step-1.5/releases/tag/v0.1.0-beta.3'>v0.1.0-beta.3</a>.</em>
